package/MDAnalysis/exceptions.py

```diff
@@ -35,6 +35,25 @@ class SelectionError(Exception):
 class NoDataError(ValueError, AttributeError):
     """Raised when empty input is not allowed or required data are missing.
 
+    This exception is raised in the following scenarios:
+
+    * Raised when a :class:`~MDAnalysis.core.topologyattrs.TopologyAttr`
+      (e.g., bonds, charges) is not present in the data.
+
+    * Raised when data is missing in an analysis class because the `run()`
+      method has not been called. Examples include polymer analysis, PSA
+      (Path Similarity Analysis), dielectric analysis, and hydrogen bond
+      analysis.
```

Comment:
```suggestion
      method has not been called, e.g. :class:`~MDAnalysis.analysis.polymer.PersistenceLength`
```

As the error is mentioned to be raised here in particular classes, it makes more sense to link to the actual class than to just reference modules.